### PR TITLE
LG-3988: Expose portalRef in components using Popover

### PR DESCRIPTION
**Pull request by stephl3**
## ✍️ Proposed changes

- expose `portalRef` in components using `Popover`

🎟 _Jira ticket:_ [LG-3988](https://jira.mongodb.org/browse/[LG-3988])
[Part 1](https://github.com/mongodb/leafygreen-ui/pull/2239) | [Part 2](https://github.com/mongodb/leafygreen-ui/pull/2241)

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: .changeset/silver-insects-change.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+'@leafygreen-ui/search-input': minor
+'@leafygreen-ui/guide-cue': minor
+'@leafygreen-ui/combobox': minor
+'@leafygreen-ui/tooltip': minor
+'@leafygreen-ui/select': minor
+'@leafygreen-ui/menu': minor
```
Comment (TheSonOfThomp) [inline .changeset/silver-insects-change.md:7]: I think we're missing:
- `data-picker`
- `number-input`
- `split-button`

Comment (stephl3) [reply .changeset/silver-insects-change.md:7]: ahh the way I was grepping led me to miss these ones. thanks for calling them out!

**Review by bruugey — COMMENTED** [review]
Can we add tests that demonstrate that the ref is properly passed around?

File: .changeset/few-geckos-think.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@leafygreen-ui/popover': minor
+'@leafygreen-ui/portal': minor
+---
+
+Manually forward `_ref` to `Portal` component and expose `portalRef` in `Popover` component
```
Comment (TheSonOfThomp) [inline .changeset/few-geckos-think.md:6]: If we decide to update the prop name in the other PR, just be sure we update this here

**Review by TheSonOfThomp — APPROVED** [review]
We should probably come up with a reusable way of doing this in the future, but this LGTM!

**Pull request merged**